pkg/controllers/sync/controller.go

```diff
@@ -717,88 +719,76 @@ func (s *SyncController) setFederatedStatus(fedResource FederatedResource, colli
 		return false, errors.Wrapf(err, "failed to update resource")
 	})
 	if err != nil {
-		runtime.HandleError(errors.Wrapf(err, "failed to update syncing annotation for %s %q", kind, name))
+		kindedKeyedLogger.Error(err, "Failed to update syncing annotation")
 		return worker.StatusError
 	}
 
 	return worker.StatusAllOK
 }
 
-func (s *SyncController) ensureDeletion(fedResource FederatedResource) worker.Result {
+func (s *SyncController) ensureDeletion(ctx context.Context, fedResource FederatedResource) worker.Result {
 	fedResource.DeleteVersions()
 
 	key := fedResource.FederatedName().String()
 	kind := fedResource.FederatedKind()
+	kindedKeyedLogger := klog.FromContext(ctx).WithValues("federated-kind", kind)
 
-	klog.V(2).Infof("Ensuring deletion of %s %q", kind, key)
+	kindedKeyedLogger.V(1).Info("Ensuring deletion of federated object")
 
 	obj := fedResource.Object()
 
 	finalizers := sets.NewString(obj.GetFinalizers()...)
 	if !finalizers.Has(FinalizerSyncController) {
-		klog.V(2).Infof("%s %q does not have the %q finalizer. Nothing to do.", kind, key, FinalizerSyncController)
+		kindedKeyedLogger.WithValues("finalizer-name", FinalizerSyncController).
+			V(2).Info("Federated object does not have the finalizer. Nothing to do")
```

Comment:
This should be v3, or we can remove the log entirely since it is a noop

Reply:
done, pls take review.